components/camera/client.go

```diff
@@ -228,6 +179,34 @@ func (c *client) Stream(
 	return stream, nil
 }
 
+func (c *client) Image(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, ImageMetadata, error) {
+	ctx, span := trace.StartSpan(ctx, "camera::client::Image")
+	defer span.End()
+	expectedType, _ := utils.CheckLazyMIMEType(mimeType)
+
+	convertedExtra, err := goprotoutils.StructToStructPb(extra)
+	if err != nil {
+		return nil, ImageMetadata{}, err
+	}
+	resp, err := c.client.GetImage(ctx, &pb.GetImageRequest{
+		Name:     c.name,
+		MimeType: expectedType,
+		Extra:    convertedExtra,
+	})
+	if err != nil {
+		return nil, ImageMetadata{}, err
+	}
+
+	if expectedType != "" && resp.MimeType != expectedType {
+		c.logger.CDebugw(ctx, "got different MIME type than what was asked for", "sent", expectedType, "received", resp.MimeType)
+	} else {
+		resp.MimeType = mimeType
+	}
+
+	resp.MimeType = utils.WithLazyMIMEType(resp.MimeType)
```

Comment:
Mime type handling seems to be making sense but just want to make sure I am understanding.

1. Use `CheckLazyMIMEType` to trim off lazy suffix if present in request.
2. Handle requested and response mimetype diff.
3. Add lazy suffix back in.

Reply:
This aligned with my read on the situation.